### PR TITLE
aws resource: `aws_iam_group_membership` and resource.SetImporter

### PR DESCRIPTION
**Pull request by talset**
provider resource: implement SetImporter to set schema.Resource.Importer when resource is not importable.

aws resource: `aws_iam_group_membership`

### CODE REVIEW

File: provider/resource.go [hunk 1]
```diff
@@ -173,6 +177,11 @@ func (r *resource) TFResource() *schema.Resource {
 	return r.tfResource
 }
 
+func (r *resource) SetImporter(i *schema.ResourceImporter) {
+	resource := r.TFResource()
```
Comment (tormath1) [inline provider/resource.go:181]:
```suggestion
	r.TFResource().Importer = i
```
`r.TFResource()` will panic if there is an issue, it does not return `error`, so we can oneline it :)

File: provider/resource.go [hunk 2]
```diff
@@ -49,6 +49,10 @@ type Resource interface {
 	// TFResource is the definition of that resource
 	TFResource() *schema.Resource
 
+	// SetImporter set schema.Resource.Importer
+	// It define the ResourceImporter implementation for this resource
```
Comment (tormath1) [inline provider/resource.go:53]:
```suggestion
	// It defines the ResourceImporter implementation for this resource
```

File: aws/cmd/functions.go [hunk 3]
```diff
@@ -501,6 +501,19 @@ var (
...
 		},
+		Function{
+			Entity: "Group",
+			Prefix: "Get",
+			FnAttributeList:       "Users",
+			SingularEntity:        "User",
+			FnPaginationAttribute: "Marker",
+			Service:               "iam",
+			Documentation: `
+			// GetGroup a list of IAM users that are in the specified IAM group
```
[the remark this answers is not in the document]
Comment (tormath1) [reply aws/cmd/functions.go:512]: `GetGroup` does not sound explicit. If I call `GetGroup` I'm expecting to get a `group` not a list of users in this group. Maybe `GetGroupUsers` ? IDK aha

**Comment by xescugc** [comment]
Can you rebase this branch with the latest `master`? We have the big update on dependencies and I think this will get affected.

**Comment by xescugc** [comment]
CI is still failing and for what I see you did not rebase with `master` yet.

**Comment by talset** [comment]
> CI is still failing and for what I see you did not rebase with `master` yet.

Didn't had time to check the ci yet (this is why I set the in progress label). But the rebase have been done.
As you can see the latest commit on master is `3e12f8d3b6ae9fcc1e6faa726f54a8ca0da43c5d` and are present in my branch (https://github.com/cycloidio/terracognita/commits/fl-iamgroupmemberships)

File: aws/resources.go [hunk 4]
```diff
@@ -1083,6 +1110,10 @@ func iamUsers(ctx context.Context, a *aws, resourceType string, filters *filter.
 }
 
 func iamUserGroupMemberships(ctx context.Context, a *aws, resourceType string, filters *filter.Filter) ([]provider.Resource, error) {
+	// if both aws_iam_group_membership and aws_iam_user_group_membership defined, keep only aws_iam_group_membership
+	if filters.IsIncluded("aws_iam_group_membership") && (!filters.IsExcluded("aws_iam_group_membership")) {
```
Comment (tormath1) [inline aws/resources.go:1114]: This condition is weird ? 
We check if the `aws_iam_group_membership` is included and if it's not excluded (so included) ?

Comment (talset) [reply aws/resources.go:1114]: Well this is something that I could check in the IsIncluded function.
But currently this function only check if all or a dedicated resource is included, not if the resource is not in exclude.

So we need to check if a user set include `aws_iam_group_membership` and exclude `aws_iam_group_membership`. If both are set, the resource will not be imported

**Comment by xescugc** [comment]
I commented to @talset to write a TODO for that specific case and open an issue to track the actual issue opened on the TF provider. So we can rely on it and remove this logic once the TF provider supports it.

File: aws/resources.go [hunk 5]
```diff
@@ -824,15 +826,42 @@ func iamGroupMemberships(ctx context.Context, a *aws, resourceType string, filte
...
+			continue
+		}
+
+		r, err := initializeResource(a, i, resourceType)
 		if err != nil {
 			return nil, err
 		}
+
+		// TODO this resource is not importable. Define our own ResourceImporter
+		// Should be removed when terraform will support it https://github.com/terraform-providers/terraform-provider-aws/pull/13795
```
Comment (xescugc) [inline aws/resources.go:851]: Create an issue also on TC and add the link in this TODO

**Pull request merged**